packages/newsroom-signup/src/Newsroom.tsx

```diff
@@ -117,6 +162,21 @@ class NewsroomComponent extends React.Component<NewsroomProps & DispatchProp<any
     },
   };
 
+  public static getDerivedStateFromProps(
+    props: NewsroomProps,
+    state: NewsroomComponentState,
+  ): NewsroomComponentState | null {
+    // @TODO/toby Confirm that when grant is rejected, it comes through as explicit `false` and not null or undefined
+    const waitingOnGrant = props.grantRequested && typeof props.grantApproved !== "boolean";
+    if (state.currentStep === STEP.PROFILE_GRANT && !waitingOnGrant) {
+      return {
+        ...state,
+        currentStep: state.currentStep + 1,
```

Comment:
I guess so, though if it doesn't save and they don't manually navigate (which triggers a save), next time they come back this will safely auto-advance them again.